src/loader.c

```diff
@@ -61,20 +82,66 @@ load_impl_library()
             fprintf(stderr, "wpe: could not load the impl library. Is there any backend installed?: %s\n", dlerror());
             abort();
         }
+        wpe_loader_set_library_name("libWPEBackend-default.so");
     }
 #endif
 
     s_impl_loader = dlsym(s_impl_library, "_wpe_loader_interface");
 }
 
+__attribute__((visibility("default")))
+void
+wpe_loader_init(const char* library_name)
+{
+#ifndef WPE_BACKEND
+    if (!library_name) {
+        fprintf(stderr, "wpe_loader_init: invalid library name\n");
+        abort();
+    }
+
+    if (s_impl_library) {
+        if (!s_impl_library_name || strcmp(s_impl_library_name, library_name) != 0) {
+            fprintf(stderr, "wpe_loader_init: already initialized\n");
+            abort();
+        }
+        return;
+    }
+
+    s_impl_library = dlopen(library_name, RTLD_NOW);
+    if (!s_impl_library) {
+        fprintf(stderr, "wpe_loader_init could not load the library '%s': %s\n", library_name, dlerror());
+        abort();
+    }
+    wpe_loader_set_library_name(library_name);
+
+    s_impl_loader = dlsym(s_impl_library, "_wpe_loader_interface");
+#endif
+}
+
+__attribute__((visibility("default")))
+const char*
+wpe_loader_get_loaded_library_name(void)
+{
+#ifdef WPE_BACKEND
+    return s_impl_library ? WPE_BACKEND : NULL;
```

Comment:
IMO this should unconditionally return WPE_BACKEND, since the same argument is used for dlopen() call when it's specified.

Reply:
But this can be called before wpe_loader_init/load_impl_library. NULL means that no library has been loaded. That's why the loaded part of the name is important.